### PR TITLE
prevent pkg-config from finding unrelated system libraries

### LINKED ISSUE #12514: Build error on mac osx (Julia 0.4 latest)
Hi all, 

I am trying to get the latest julia build for my mac but I can't seem to compile it. Near the end of the compilation I get the following errors
```
[ 22%] Building C object CMakeFiles/git2.dir/src/unix/map.c.o
[ 22%] Building C object CMakeFiles/git2.dir/src/unix/realpath.c.o
[ 22%] Building C object CMakeFiles/git2.dir/deps/http-parser/http_parser.c.o
Linking C shared library libgit2.dylib
Undefined symbols for architecture x86_64:
  "_libiconv", referenced from:
      _git_path_iconv in path.c.o
  "_libiconv_close", referenced from:
      _git_path_direach in path.c.o
      _git_path_iconv_clear in path.c.o
      _git_path_diriter_free in path.c.o
      _git_path_dirload in path.c.o
  "_libiconv_open", referenced from:
      _git_path_direach in path.c.o
      _git_path_iconv_init_precompose in path.c.o
      _git_path_diriter_init in path.c.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[4]: *** [libgit2.0.23.1.dylib] Error 1
make[3]: *** [CMakeFiles/git2.dir/all] Error 2
make[2]: *** [all] Error 2
make[1]: *** [libgit2/build/libgit2.dylib] Error 2
make: *** [julia-deps] Error 2
```
Looks like a linker error. But am not sure what is causing this. I am just running the default make command. I am on OSX Yosemite.

Any clues?

thanks,

### PR DESCRIPTION
**Pull request by vtjnash**
fix #12514

### CODE REVIEW

**Comment by tkelman** [comment]
Hm, ref #10010, it looks like this fails to find openssl in multilib compilation of 32 bit linux from 64 bit linux. That's better than a build/link failure, but may make testing https + libgit2 on 32 bit linux a little hard in the future. I can't seem to figure out how to get apt-get to install `libssl-dev:i386` without uninstalling `build-essential` on ubuntu 14.04, so maybe we should go with this change anyway.

**Comment by vtjnash** [comment]
yeah, its likely this would break distributions, which may actually want to be using pkg-config (and would thus force them to unset / unexport this variable). failing to find openssl may actually be good, since if we're building generic linux binaries, then we really only want to find libs that will actually be on all target machines.

**Comment by tkelman** [comment]
Not quite. We bundle the openssl that is on the build machine. Ref #10763. No https on Linux without it, unfortunately. While there will almost always be a system version present on pretty much any Linux distribution, it's not guaranteed to be API or ABI compatible with the version we build libgit2 against on the buildbots. At least libgit2 can use native https on OS X and Windows.

@nalimilan do you need this pkg-config variable to be set while building julia? Will this trigger an rpmlint warning or anything?

**Comment by nalimilan** [comment]
Am I correct that it will break `USE_SYSTEM_LIBGIT2=1` if not unsetting `PKG_CONFIG_LIBDIR` manually? Why not make it conditional then?

**Comment by tkelman** [comment]
How would this break `USE_SYSTEM_LIBGIT2`? Enabling that just means we don't build libgit2 and instead rely on `ccall` to find it somewhere on the path, pretty basic in this case.

**Comment by nalimilan** [comment]
Ah, right, I forgot that we don't actually link to libgit2. So should be fine with me.

File: Make.inc [hunk 1]
```diff
@@ -160,6 +160,9 @@ BUILD_LLDB = 0
 # Path to cmake (override in Make.user if needed)
 CMAKE ?= cmake
 
+# Point pkg-config to only look at our libraries
+export PKG_CONFIG_LIBDIR = $(JULIAHOME)/usr/lib/pkgconfig
+
```
Comment (stevengj) [inline Make.inc]: Does this work?  http://stackoverflow.com/questions/23843106/how-to-set-environment-variable-in-makefile

Comment (stevengj) [inline Make.inc]: Oh, never mind, I see that that answer references the `export` command in `make`; I never knew about this one.

**Comment by vtjnash** [comment]
let's try this for a bit. if it turns out we need some `.pc` file, we can add rules to copy in the right ones from the right sysroot.

**Comment by nalimilan** [comment]
Actually this does break when using `USE_SYSTEM_LIBGIT2=0 USE_SYSTEM_SSH2=1`. In that case libgit2's CMake script doesn't detect libssh2 using pkg-config, so SSH is disabled.

**Pull request merged**